asset-transfer-basic/chaincode-javascript/lib/assetTransfer.js

```diff
@@ -56,100 +56,93 @@ class AssetTransfer extends Contract {
+        const updatedAsset = {
             ID: id,
             Color: color,
             Size: size,
             Owner: owner,
             AppraisedValue: appraisedValue,
         };
-
         return ctx.stub.putState(id, Buffer.from(JSON.stringify(updatedAsset)));
     }
 
-    // deleteAsset deletes an given asset from the world state.
-    async deleteAsset(ctx, id) {
-        const exists = await this.assetExists(ctx, id);
+    // DeleteAsset deletes an given asset from the world state.
+    async DeleteAsset(ctx, id) {
+        const exists = await this.AssetExists(ctx, id);
         if (!exists) {
             throw new Error(`The asset ${id} does not exist`);
         }
-
         return ctx.stub.deleteState(id);
     }
 
-    // assetExists returns true when asset with given ID exists in world state.
-    async assetExists(ctx, id) {
+    // AssetExists returns true when asset with given ID exists in world state.
+    async AssetExists(ctx, id) {
         const assetJSON = await ctx.stub.getState(id);
-        if (!assetJSON || assetJSON.length === 0) {
-            return false;
-        }
-        return true;
+        return assetJSON && assetJSON.length > 0;
     }
 
-    // transferAsset updates the owner field of asset with given id in the world state.
-    async transferAsset(ctx, id, newOwner) {
-        let assetString = await this.readAsset(ctx, id);
-
-        let asset = JSON.parse(assetString);
+    // TransferAsset updates the owner field of asset with given id in the world state.
+    async TransferAsset(ctx, id, newOwner) {
+        const assetString = await this.ReadAsset(ctx, id);
+        const asset = JSON.parse(assetString);
         asset.Owner = newOwner;
-
-        await ctx.stub.putState(id, Buffer.from(JSON.stringify(asset)));
+        return ctx.stub.putState(id, Buffer.from(JSON.stringify(asset)));
     }
 
-    // getAllAssets returns all assets found in the world state.
-    async getAllAssets(ctx) {
+    // GetAllAssets returns all assets found in the world state.
+    async GetAllAssets(ctx) {
         const allResults = [];
         // range query with empty string for startKey and endKey does an open-ended query of all assets in the chaincode namespace.
-        for await (const { key, value } of ctx.stub.getStateByRange("", "")) {
-            const strValue = Buffer.from(value).toString('utf8');
+        const iterator = await ctx.stub.getStateByRange('', '');
+        let result = await iterator.next();
+        while (!result.done) {
+            const strValue = Buffer.from(result.value.value.toString()).toString('utf8');
             let record;
             try {
                 record = JSON.parse(strValue);
             } catch (err) {
                 console.log(err);
                 record = strValue;
             }
-            allResults.push({ Key: key, Record: record });
+            allResults.push({ Key: result.value.key, Record: record });
```

Comment:
btw in chaincode-go , getALL returns []asset,  no Key & Record

Reply:
Will update